cmd/cloud-api-adaptor/main.go

```diff
@@ -139,6 +140,7 @@ func (cfg *daemonConfig) Setup() (cmd.Starter, error) {
 var config cmd.Config = &daemonConfig{}
 
 func main() {
+	go probe.Start()
```

Comment:
What happens if the probe.Start fails ? Does CAA needs to bail out as well?
Also if CAA fails to start due to an error post starting the probe, does the goroutine needs to be killed ?

Reply:
@bpradipt no, we can change the func name for free.

Reply:
failed startupProbe will cause pods recreating. per https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
```
If the startup probe never succeeds, the container is killed after 300s and subject to the pod's restartPolicy
```

Reply:
It'll be good to add a comment in the code, to help new devs understand why error from the goroutine is not handled. 
Like "We don't handle error from the probe goroutine since if the goroutine fails to start, the probe configured as part of the deployment will fail and the CAA containerwill be killed and restarted"